public/js/src/module/store.js

```diff
@@ -304,32 +307,42 @@ const surveyStore = {
                         .filter( existing => binaryDefaultKeys.indexOf( existing ) < 0 );
                 }
 
-                const update = Object.assign( {}, _transformExternalDataToXmlStr( survey ), {
-                    // Note: if survey.resources = undefined/null, do not store empty array
-                    // as it means there are no resources to store (and load) - they may be added later by form-cache.js
-                    resources: survey.resources ? resourceKeys : undefined,
-                    binaryDefaults: survey.binaryDefaults ? binaryDefaultKeys : undefined,
-                } );
+                let update = Object.assign( {}, _serializeSurvey( survey ) );
+
+                // Note: if survey.resources = undefined/null, do not store empty array
+                // as it means there are no resources to store (and load) - they may be added later by form-cache.js
+                if ( survey.resources ) {
+                    update.resources = resourceKeys;
+                }
+
+                if ( survey.binaryDefaults ) {
+                    update.resources = binaryDefaultKeys;
+                }
```

Comment:
Oof, good copypasta catch! This change is primarily to avoid adding unnecessary properties now that `store.survey.update` returns the updated IndexedDB result rather than the incoming `survey` argument (these properties broke a lot of tests unnecessarily and would have introduced more noise), and partly stylistic. Of course not everyone agrees, but it's _generally_ idiomatic in JS to treat `null` as an explicit value and `undefined` as an absence of a property or return value. It's also generally idiomatic to avoid referencing `undefined` as a value because ~~everything is terrible~~ [it's not a reserved keyword](https://eslint.org/docs/rules/no-undefined).